dsgrid/config/project_config.py

```diff
@@ -458,15 +474,21 @@ def load_dimensions(self, dimension_manager: DimensionRegistryManager):
         supplemental_dimensions = dimension_manager.load_dimensions(
             self.model.dimensions.supplemental_dimension_references
         )
-        dims = list(itertools.chain(base_dimensions.values(), supplemental_dimensions.values()))
+        self._base_dimensions.update(base_dimensions)
+        self._supplemental_dimensions.update(supplemental_dimensions)
+
+        dims = []
+        dims_by_type = defaultdict(list)
+        for dim in self.iter_dimensions():
+            dims.append(dim)
+            dims_by_type[dim.model.dimension_type].append(dim)
         check_uniqueness((x.model.name for x in dims), "dimension name")
```

Comment:
Do we really need to check uniqueness here? (Seems like that would be done during registration/update and wouldn't need to be repeated?)

Reply:
We've been checking for name uniqueness for a very long time here. My memory is a little fuzzy, but I think this is a place where everything is complete. I agree that it seems weird.